analyzers/src/SonarAnalyzer.CSharp/Rules/UseParamsForVariableArguments.cs

```diff
@@ -26,37 +26,44 @@ public sealed class UseParamsForVariableArguments : SonarDiagnosticAnalyzer
 
-        private static readonly DiagnosticDescriptor rule =
+        private static readonly DiagnosticDescriptor Rule =
             DescriptorFactory.Create(DiagnosticId, MessageFormat);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterNodeAction(
                 c =>
                 {
-                    var methodDeclaration = (MethodDeclarationSyntax)c.Node;
-
-                    if (methodDeclaration.Identifier.IsMissing ||
-                        methodDeclaration.Modifiers.Any(SyntaxKind.ExternKeyword) ||
-                        !HasAnyArgListParameter(methodDeclaration))
-                    {
-                        return;
-                    }
-
-                    var methodSymbol = c.SemanticModel.GetDeclaredSymbol(methodDeclaration);
-                    if (methodSymbol != null &&
-                        methodSymbol.IsPubliclyAccessible() &&
-                        methodSymbol.GetInterfaceMember() == null &&
-                        methodSymbol.OverriddenMethod == null)
+                    if (c.Node.ParameterList() is { Parameters: { Count: > 0 } parameters }
+                        && parameters.Any(p => p.Identifier.IsKind(SyntaxKind.ArgListKeyword))
+                        && CheckModifiers(c.Node)
+                        && c.Node.GetIdentifier() is { IsMissing: false } identifier
+                        && MethodSymbol(c.Node, c.SemanticModel) is { } methodSymbol
+                        && !methodSymbol.IsOverride
+                        && methodSymbol.IsPubliclyAccessible()
+                        && methodSymbol.GetInterfaceMember() == null)
                     {
-                        c.ReportIssue(Diagnostic.Create(rule, methodDeclaration.Identifier.GetLocation()));
+                        c.ReportIssue(Diagnostic.Create(Rule, identifier.GetLocation()));
                     }
                 },
-                SyntaxKind.MethodDeclaration);
-        }
+                SyntaxKind.MethodDeclaration,
+                SyntaxKind.ConstructorDeclaration,
+                SyntaxKind.ClassDeclaration,
+                SyntaxKind.StructDeclaration,
+                SyntaxKindEx.RecordClassDeclaration,
+                SyntaxKindEx.RecordStructDeclaration);
+
+        private static IMethodSymbol MethodSymbol(SyntaxNode node, SemanticModel semanticModel) =>
+            node switch
+            {
+                BaseMethodDeclarationSyntax method => semanticModel.GetDeclaredSymbol(method),
+                TypeDeclarationSyntax type => type.PrimaryConstructor(semanticModel),
+                _ => null,
```

Comment:
Based on the syntax kinds that we are registering for, they should all be covered with BaseMethodDeclaration and TypeDeclaration, so maybe this defensive switch case is not needed and you can turn it into a ternary operator and reach 100% coverage

Reply:
100% coverage at the expense of a runtime exception? No thanks. (Funny: This PR has more "real coverage than most of the others, but to reach the percentage goal you have to make it less robust).

Also: this switch is easier to extend in the future.

Reply:
That's true, and encountering AD0001s can be beneficial as it allows us to identify and address new scenarios. While the default option provides a safety net, we will remain unaware of any new cases to cover